From 77051e16b50d0d718a0db8e0b9233b945f6850d8 Mon Sep 17 00:00:00 2001
From: Chaitanya Kulkarni <kch@nvidia.com>
Date: Tue, 7 Nov 2023 13:59:12 -0800
Subject: [PATCH 1/2] nvme-fabrics: don't repeat the code

In nvmf_map_queues() while mapping separate read/write queues and shared
read/write queues we set HCTX_TYPE_DEFAULT nr_queues to io_queues
HCTX_TYPE_DEFAULT and HCTX_TYPE_DEFAULT queue offset to 0. This code
is repeated for both above mentioned cases.

Move the common code out of both cases.

Signed-off-by: Chaitanya Kulkarni <kch@nvidia.com>
---
 drivers/nvme/host/fabrics.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/nvme/host/fabrics.c b/drivers/nvme/host/fabrics.c
index 4673ead69c5f..98b33664d87c 100644
--- a/drivers/nvme/host/fabrics.c
+++ b/drivers/nvme/host/fabrics.c
@@ -1095,20 +1095,17 @@ void nvmf_map_queues(struct blk_mq_tag_set *set, struct nvme_ctrl *ctrl,
 {
 	struct nvmf_ctrl_options *opts = ctrl->opts;
 
+	set->map[HCTX_TYPE_DEFAULT].nr_queues = io_queues[HCTX_TYPE_DEFAULT];
+	set->map[HCTX_TYPE_DEFAULT].queue_offset = 0;
+
 	if (opts->nr_write_queues && io_queues[HCTX_TYPE_READ]) {
 		/* separate read/write queues */
-		set->map[HCTX_TYPE_DEFAULT].nr_queues =
-			io_queues[HCTX_TYPE_DEFAULT];
-		set->map[HCTX_TYPE_DEFAULT].queue_offset = 0;
 		set->map[HCTX_TYPE_READ].nr_queues =
 			io_queues[HCTX_TYPE_READ];
 		set->map[HCTX_TYPE_READ].queue_offset =
 			io_queues[HCTX_TYPE_DEFAULT];
 	} else {
 		/* shared read/write queues */
-		set->map[HCTX_TYPE_DEFAULT].nr_queues =
-			io_queues[HCTX_TYPE_DEFAULT];
-		set->map[HCTX_TYPE_DEFAULT].queue_offset = 0;
 		set->map[HCTX_TYPE_READ].nr_queues =
 			io_queues[HCTX_TYPE_DEFAULT];
 		set->map[HCTX_TYPE_READ].queue_offset = 0;
-- 
2.40.0

